src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2790,14 +2790,17 @@ namespace DictValues {
 static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
     ASRUtils::require_impl(x.n_args == 1, "Call to dict.values must have no argument",
         x.base.base.loc, diagnostics);
-    ASRUtils::require_impl(ASR::is_a<ASR::Dict_t>(*ASRUtils::expr_type(x.m_args[0])),
+    ASR::ttype_t *dict_type = ASR::is_a<ASR::Const_t>(*ASRUtils::expr_type(x.m_args[0]))
+                                  ? ASRUtils::type_get_past_const(ASRUtils::expr_type(x.m_args[0]))
+                                  : ASRUtils::expr_type(x.m_args[0]);
+    ASRUtils::require_impl(ASR::is_a<ASR::Dict_t>(*dict_type),
         "Argument to dict.values must be of dict type",
         x.base.base.loc, diagnostics);
     ASRUtils::require_impl(ASR::is_a<ASR::List_t>(*x.m_type) &&
-        ASRUtils::check_equal_type(ASRUtils::get_contained_type(x.m_type),
-        ASRUtils::get_contained_type(ASRUtils::expr_type(x.m_args[0]), 1)),
-        "Return type of dict.values must be of list of dict value element type",
-        x.base.base.loc, diagnostics);
+                               ASRUtils::check_equal_type(ASRUtils::get_contained_type(x.m_type),
```

Comment:
Please remove all the formatting changes in this PR.
We limit the source code to 80 characters per line (more or less 10 characters), so do follow that style.

Reply:
I am sorry for this. The IDE did this on it's own.